### PR TITLE
[Image] Components dialog validation not applied for inactive tabs

### LINKED ISSUE #1974: [Image] Components dialog validation not applied for inactive tabs
In the image core component the metadata tab has required fields. A user can leave these blank but is still able to submit the dialog from the Asset tab. The required fields on inactive tabs are not checked for completion when the user saves the dialog.

Expected Behaviour: The user should not be able to save the dialog. The dialog should left be open with a warning on the tab that has required filed which are left blank.

SITES-4820

### PR DESCRIPTION
**Pull request by bpauli**
- add custom validation for alt text and decorative combination
- update UI in case checkbox for decorative is checked

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #1974 
| Patch: Bug Fix?          |
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

File: content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js [hunk 1]
```diff
@@ -143,10 +143,25 @@
 
         }
 
-        $(window).adaptTo("foundation-registry").register("foundation.validation.selector", {
-            submittable: ".cmp-image__editor-alt-text",
-            candidate: ".cmp-image__editor-alt-text:not(:hidden)",
-            exclusion: ".cmp-image__editor-alt-text *"
+        $(window).adaptTo("foundation-registry").register("foundation.validation.validator", {
```
Comment (vladbailescu) [inline content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js:146]: Wondering if we could move this validator to a commons script and reuse it between the two versions.

**Review by jckautzmann — APPROVED** [review]

File: content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js (hunk 1 above)
Comment (bpauli) [reply content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js]: Indeed, makes sense. Please check 2c1016387d4002473a9e72270b77cae460ef8367

File: content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js [hunk 2]
```diff
@@ -46,6 +47,7 @@
     var altTextFromDAM;
     var altCheckboxSelector = "coral-checkbox[name='./altValueFromDAM']";
     var altInputSelector = "input[name='./alt']";
+    var decorativeCheckboxSelector = "coral-checkbox[name='./isDecorative']"
```
Comment (vladbailescu) [inline content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js:50]: Missing semicolon

File: content/src/content/jcr_root/apps/core/wcm/components/commons/v1/clientlibs/editor/decorativeAltTextValidator/decorativeAltTextValidator.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * Creates a validator for an alt text which is required if decorative checkbox is not checked
+     *
+     * @param {String} decorativeCheckboxSelector the selector for the decorative checkbox in the dialog
+     * @param {String} altTextFieldSelector the selector for the alt text in the dialog
+     */
+    window.CQ.CoreComponents.DecorativeAltTextValidator.v1 = function(decorativeCheckboxSelector, altTextFieldSelector) {
+
+        let self = this;
```
Comment (vladbailescu) [inline content/src/content/jcr_root/apps/core/wcm/components/commons/v1/clientlibs/editor/decorativeAltTextValidator/decorativeAltTextValidator.js:31]: I don't think `let` is allowed in linting.

**Pull request merged**